core/src/main/java/jenkins/security/ResourceDomainRootAction.java

```diff
@@ -291,10 +291,12 @@ private static Token decode(String value) {
                     throw new IllegalArgumentException("Failed mac check for " + rest);
                 }
 
-                String[] splits = rest.split(":", 3);
-                String authenticationName = Util.fixEmpty(splits[0]);
-                String epoch = splits[1];
-                String browserUrl = splits[2];
+                String[] splits = rest.split("[:]", 3);
+                String epoch = splits[0];
+                int authenticationNameLength = Integer.parseInt(splits[1]);
+                String authenticationNameAndBrowserUrl = Util.fixEmpty(splits[2]);
+                String authenticationName = authenticationNameAndBrowserUrl.substring(0, authenticationNameLength);
```

Comment:
Is letting `authenticationNameAndBrowserUrl` be null and doing a nullptr dereference here intentional?